src/completions.jl

```diff
@@ -0,0 +1,261 @@
+import JuliaLowering: SyntaxTree, JuliaLowering as JL
+import JuliaSyntax: @K_str, @KSet_str, JuliaSyntax as JS
+
+# get_fileinfo(s::ServerState, t::TextDocumentPositionParams) = get_fileinfo(s, t.textDocument)
+get_fileinfo(s::ServerState, t::CompletionParams) = get_fileinfo(s, t.textDocument)
+
+"""
+Like `Base.unique`, but over node ids, and with this comment promising that the
+lowest-index copy of each node is kept.
+"""
+function unique(sl::JL.SyntaxList)
```

Comment:
```suggestion
function unique_syntax_list(sl::JL.SyntaxList)
```
It's a bit tricky, but if we don't use `Base.unique`, this defines a new `unique` generic function, which will shadow `Base.unique` in this module namespace. So, either explicitly extend `Base.unique` or just give it a different name. I'd recommend the later since this function is very specific to this package.